extensions/blocks/slideshow/slideshow.php

```diff
@@ -24,5 +24,88 @@
  */
 function jetpack_slideshow_block_load_assets( $attr, $content ) {
 	Jetpack_Gutenberg::load_assets_as_required( 'slideshow' );
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		return jetpack_slideshow_block_render_amp( $attr );
+	}
 	return $content;
 }
+
+/**
+ * Render slideshow block for AMP
+ *
+ * @param array $attr Array containing the slideshow block attributes.
+ *
+ * @return string
+ */
+function jetpack_slideshow_block_render_amp( $attr ) {
+	global $wp_block_jetpack_slideshow_id;
+	$wp_block_jetpack_slideshow_id = ( $wp_block_jetpack_slideshow_id || 0 ) + 1;
+	$amp_carousel_id               = sprintf(
+		'wp-block-jetpack-slideshow__%s',
+		intval( $wp_block_jetpack_slideshow_id )
+	);
+
+	$ids        = empty( $attr['ids'] ) ? array() : $attr['ids'];
+	$autoplay   = empty( $attr['autoplay'] ) ? false : $attr['autoplay'];
+	$autoplay   = false;
+	$delay      = empty( $attr['delay'] ) ? 3 : intval( $attr['delay'] );
+	$align      = isset( $attr['align'] ) ? $attr['align'] : 'center';
+	$classes    = array(
+		'wp-block-jetpack-slideshow',
+		'wp-amp-block',
+		'align' . $align,
+	);
+	$slides     = array_map(
+		function( $id ) {
+			$caption    = wp_get_attachment_caption( $id );
+			$src        = wp_get_attachment_image_src( $id, 'full' );
+			$figcaption = $caption ? sprintf(
+				'<figcaption class="wp-block-jetpack-slideshow_caption gallery-caption">%s</figcaption>',
+				wp_kses_post( $caption )
+			) : '';
+			$amp_img    = sprintf(
+				'<amp-img src="%s" width="%s" height="%s" alt="%s" class="wp-block-jetpack-slideshow_image" />',
```

Comment:
Might be better to just use `wp_get_attachment_image()` here and let the image sanitizer do the conversion to `amp-img`. This would include the `srcset` automatically.

Also, an aside: it would need to be `<amp-img></amp-img>` not self-closing as `<amp-img />`.

Reply:
Good idea: https://github.com/Automattic/jetpack/pull/13009/commits/d59da6ce12f0cb3775e5795eb767da6665088714

FWIW, the old way was functioning ok self-closing. Would the sanitizer have caught and fixed that as well?

Reply:
It probably was handled automatically by the libxml in PHP, but outside of that context it would have caused problems.